gix-transport/src/client/blocking_io/http/reqwest/remote.rs

```diff
@@ -44,38 +44,44 @@ impl Default for Remote {
+                    .connect_timeout(std::time::Duration::from_secs(20))
+                    .http1_title_case_headers()
+                    .redirect(reqwest::redirect::Policy::custom({
+                        move |attempt| {
+                            if allow_redirects.load(atomic::Ordering::Relaxed) {
+                                let curr_url = attempt.url();
+                                let prev_urls = attempt.previous();
 
-                            match prev_urls.first() {
-                                Some(prev_url) if prev_url.host_str() != curr_url.host_str() => {
-                                    // git does not want to be redirected to a different host.
-                                    attempt.stop()
-                                }
-                                _ => {
-                                    // emulate default git behaviour which relies on curl default behaviour apparently.
-                                    const CURL_DEFAULT_REDIRS: usize = 50;
-                                    if prev_urls.len() >= CURL_DEFAULT_REDIRS {
-                                        attempt.error("too many redirects")
-                                    } else {
-                                        attempt.follow()
+                                match prev_urls.first() {
+                                    Some(prev_url) if prev_url.host_str() != curr_url.host_str() => {
+                                        // git does not want to be redirected to a different host.
+                                        attempt.stop()
+                                    }
+                                    _ => {
+                                        // emulate default git behaviour which relies on curl default behaviour apparently.
+                                        const CURL_DEFAULT_REDIRS: usize = 50;
+                                        if prev_urls.len() >= CURL_DEFAULT_REDIRS {
+                                            attempt.error("too many redirects")
+                                        } else {
+                                            attempt.follow()
+                                        }
                                     }
                                 }
+                            } else {
+                                attempt.stop()
                             }
-                        } else {
-                            attempt.stop()
                         }
-                    }
-                }))
+                    }))
+            }
+
+            // We may error while configuring, which is expected as part of the internal protocol. The error will be
+            // received and the sender of the request might restart us.
+            let client_ssl_verify = setup_client_builder(allow_redirects.clone()).build()?;
+
+            let client_no_ssl_verify = setup_client_builder(allow_redirects.clone())
```

Comment:
(*There is no need to create two clients for this. Instead, just re-assign the (one) client conditionally - this note is for reference only as reqwest support has to be removed for now.*)

Reply:
I am not sure I understood what you meant. I did it this way, because the config is read from each Request after creating the client, so instead of recreating clients every time I did it like this. However, I removed reqwest support for now.